builder/bundles.go

```diff
@@ -436,17 +436,43 @@ func genUpdateBundleSpecialFiles(chrootDir string, b *Builder) error {
 	return ioutil.WriteFile(filepath.Join(swupdDir, "format"), []byte(b.State.Mix.Format), 0644)
 }
 
+func downloadRpms(packagerCmd, rpmList []string, baseDir string, maxRetries int) error {
+	var err error
+
+	args := merge(packagerCmd, "--installroot="+baseDir, "install", "--downloadonly")
+	args = append(args, rpmList...)
+
+	for retries := 0; retries < maxRetries; retries++ {
+		_, err = helpers.RunCommandOutputEnv(args[0], args[1:], []string{"LC_ALL=en_US.UTF-8"})
+		if err == nil {
+			return err
+		}
+		fmt.Printf("RPM download attempt %d failed. Retrying up to %d times.\n", retries+1, maxRetries)
+	}
+	return err
+}
+
 func installBundleToFull(packagerCmd []string, buildVersionDir string, bundle *bundle) error {
 	var err error
 	baseDir := filepath.Join(buildVersionDir, "full")
-	args := merge(packagerCmd, "--installroot="+baseDir, "install")
+	rpmList := []string{}
+
 	if len(bundle.AllPackages) > 0 {
 		// There were packages directly included for this bundle so
 		// install to full chroot. This check is necessary so we don't
 		// call dnf install with no package listed.
 		for p := range bundle.AllPackages {
-			args = append(args, p)
+			rpmList = append(rpmList, p)
+		}
+
+		// Retry RPM downloads to avoid timeout failures due to slow network
+		err = downloadRpms(packagerCmd, rpmList, baseDir, 3)
```

Comment:
It would be nicer to have a constant defined with the number of retries instead of having a magic number on the function call. I also think it would be nice if this parameter could be set via command line, so users with troublesome connection could thinker with a better value.

Reply:
Ok, I can add this.